### PR TITLE
Remote Environment Controller implementation

### LINKED ISSUE #416: Create a new Kyma component for RE operator
Confirm these statements before you submit the issue:

- [x] I have searched open and closed issues for duplicates.
- [x] I have read the contributing guidelines.
---

**Description**

We should prepare 'kyma-scaffolding' for our RE operator. Also, we need to productise our PoC results on this topic.

PoC: https://github.com/janmedrek/kyma/tree/kubebuilder-poc

Acceptance:
- Helm chart for operator prepared
- Jenkinsfile created
- Tests complemented
- RE chart copied (for now) to operator directory

### PR DESCRIPTION
**Pull request by Szymongib**
**Description**

Changes proposed in this pull request:

- Implemented new component - Remote Environment Controller

**Related issue(s)** 
Part of #416

### CODE REVIEW

File: components/remote-environment-controller/pkg/controller/reconciler.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+func (r *remoteEnvironmentReconciler) installNewREChart(name string) {
+	_, err := r.helmClient.InstallReleaseFromChart(reChartDirectory, r.namespace, name, r.overrides)
+	if err != nil {
+		log.Error("Error while installing release for %s RE: %s", name, err.Error())
+	} else {
+		log.Infof("New Helm chart for %s Remote Environment installed", name)
+	}
+}
+
+// TODO: consider returning error to requeue the request
```
Comment (janmedrek) [inline components/remote-environment-controller/pkg/controller/reconciler.go]: Do we need to requeue delete request? Seems like tiller.UninstallRelease RPC does it for us, please double-check that.

File: components/remote-environment-controller/pkg/kymahelm/client.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+	helm *helm.Client
+}
+
+func NewClient(host string) HelmClient {
+	return &helmClient{
+		helm: helm.NewClient(helm.Host(host)),
+	}
+}
+
+// ListReleases .
```
Comment (janmedrek) [inline components/remote-environment-controller/pkg/kymahelm/client.go]: Redundant commentary in this file :)

File: components/remote-environment-controller/pkg/controller/reconciler.go (hunk 1 above)
Comment (Szymongib) [reply components/remote-environment-controller/pkg/controller/reconciler.go]: Removed comment

File: components/remote-environment-controller/pkg/kymahelm/client.go (hunk 2 above)
Comment (Szymongib) [reply components/remote-environment-controller/pkg/kymahelm/client.go]: Done

File: components/remote-environment-controller/pkg/controller/reconciler.go [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+		return reconcile.Result{}, err
+	}
+
+	releaseExist, err := r.checkReleaseExistence(request.Name)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if releaseExist {
+		// Handles cases: RE with chart exist on controller startup, RE is updated, full RE chart is installed
```
[the remark this answers is not in the document]
Comment (Szymongib) [reply components/remote-environment-controller/pkg/controller/reconciler.go]: It might not be clear enough if someone did not work with Kubebuilder

**Pull request merged**